state-sync/state-sync-v2/state-sync-driver/src/notification_handlers.rs

```diff
@@ -181,41 +146,48 @@ impl ConsensusNotificationHandler {
+                if let Some(consensus_sync_request) = consensus_sync_request {
                     self.respond_to_sync_notification(
-                        sync_request.consensus_sync_notification,
+                        consensus_sync_request.consensus_sync_notification,
                         Ok(()),
                     )
-                    .await;
+                    .await?;
                 }
                 return Ok(());
             }
 
-            // Check if the sync deadline has been exceeded (timed out since the last commit)
-            let max_time_between_commits = Duration::from_millis(CONSENSUS_SYNC_REQUEST_TIMEOUT_MS);
-            let next_commit_deadline = sync_request
-                .last_commit_timestamp
-                .checked_add(max_time_between_commits)
-                .ok_or_else(|| {
-                    Error::IntegerOverflow("The new commit deadline has overflown!".into())
-                })?;
-            if SystemTime::now()
-                .duration_since(next_commit_deadline)
-                .is_ok()
-            {
-                // TODO(joshlind): log this!
-
-                // Remove the sync request and notify consensus that the request timed out
-                if let Some(sync_request) = self.sync_request.take() {
-                    self.respond_to_sync_notification(
-                        sync_request.consensus_sync_notification,
-                        Err(Error::UnexpectedError(
-                            "Sync request timed out! Hit the max commit time!".into(),
-                        )),
-                    )
-                    .await;
+            // Check if the commit deadline has been exceeded (timed out since the last commit)
+            let consensus_sync_request = self.get_consensus_sync_request().lock().take();
+            if let Some(consensus_sync_request) = consensus_sync_request {
+                let max_time_between_commits =
+                    Duration::from_millis(CONSENSUS_SYNC_REQUEST_TIMEOUT_MS);
+                let next_commit_deadline = consensus_sync_request
+                    .get_last_commit_timestamp()
+                    .checked_add(max_time_between_commits)
+                    .ok_or_else(|| {
+                        Error::IntegerOverflow("The new commit deadline has overflown!".into())
+                    })?;
+                if SystemTime::now()
+                    .duration_since(next_commit_deadline)
+                    .is_ok()
+                {
+                    // Remove the sync request and notify consensus that the request timed out
+                    let error = Error::UnexpectedError(format!(
+                        "Sync request timed out! Hit the max time between commits: {:?}",
+                        max_time_between_commits
+                    ));
+                    let consensus_sync_request = self.get_consensus_sync_request().lock().take();
```

Comment:
How exactly do you take the request twice?  Am I missing something?

Reply:
Copy-pasta 😄 Fixed! Thanks for finding these @gregnazario! Better now than debugging things in the unit test 😉